proto/structured.proto

```diff
@@ -27,61 +27,51 @@ option (gogoproto.sizer_all) = true;
 option (gogoproto.marshaler_all) = true;
 option (gogoproto.unmarshaler_all) = true;
 
-message Table {
-  optional string name = 1 [(gogoproto.nullable) = false];
+// For Cockroach internal use.
+message Internal {
+  optional uint32 id = 1 [(gogoproto.nullable) = false];
 }
-
+ 
 message Column {
   enum ColumnType {
     BYTES = 0;
   }
-
   optional string name = 1 [(gogoproto.nullable) = false];
   optional ColumnType type = 2 [(gogoproto.nullable) = false];
+  // For Cockroach internal use.
+  optional Internal internal = 9 [(gogoproto.nullable) = true];
 }
 
 message Index {
   optional string name = 1 [(gogoproto.nullable) = false];
   optional bool unique = 2 [(gogoproto.nullable) = false];
+  // An ordered list of column names of which the index is comprised. Each
+  // column_name refers to a column in the TableSchema's columns. Special
+  // care is taken to update this when deleting columns.
+  repeated string column_names = 3 [(gogoproto.nullable) = false];
+  // For Cockroach internal use.
+  optional Internal internal = 9 [(gogoproto.nullable) = true];
```

Comment:
we talked about this a bit - the benefit of having the internal message is that it can be made nullable, which is slightly more semantically correct when set from a client (i.e. the `internal` thing will be completely absent, rather than initialized to a zero value).